fontbe/src/orchestration.rs

```diff
@@ -219,8 +220,99 @@ impl From<&VariationRegion> for TupleBuilder {
     }
 }
 
+#[repr(u8)]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum LocaFormat {
+    Short,
+    Long,
+}
+
+impl LocaFormat {
+    pub fn new(loca: &[u32]) -> LocaFormat {
+        // https://github.com/fonttools/fonttools/blob/1c283756a5e39d69459eea80ed12792adc4922dd/Lib/fontTools/ttLib/tables/_l_o_c_a.py#L37
+        if loca.last().copied().unwrap_or_default() < 0x20000
+            && loca.iter().all(|offset| offset % 2 == 0)
+        {
+            LocaFormat::Short
+        } else {
+            LocaFormat::Long
+        }
+    }
+}
+
+impl TryFrom<u8> for LocaFormat {
+    type Error = ();
+
+    fn try_from(value: u8) -> Result<Self, ()> {
+        match value {
+            _ if value == LocaFormat::Short as u8 => Ok(LocaFormat::Short),
+            _ if value == LocaFormat::Long as u8 => Ok(LocaFormat::Short),
+            _ => Err(()),
+        }
+    }
+}
+
+// Free function of specific form to fit macro
+fn loca_format_from_file(file: &Path) -> LocaFormat {
+    let bytes = fs::read(file).unwrap();
+    LocaFormat::try_from(*bytes.first().unwrap()).unwrap()
```

Comment:
since we really aren't using the error value here at all I would be more tempted to remove `TryFrom` and just do,

```suggestion
    match bytes.first() {
        Some(0) => LocaFormat::Short,
        Some(1) => LocaFormat::Long,
        _ => panic!("serialized LocaFormat is invalid"),
    }
```

 which might be a bit easier to follow and a bit easier to debug if anything actually *does* go wrong, while also being less code.

Reply:
That does seem easier to read, will change.